finalProject/index.js

```diff
@@ -0,0 +1,189 @@
+    `;
+    recipeGrid.appendChild(recipeCard);
+  });
+}
+
+displayRecipes(recipes);
+
+// Add New Recipe
+ document.getElementById("addRecipeForm").addEventListener("submit", function (e) {
+  e.preventDefault();
+
+  const title = document.getElementById("recipeTitle").value;
+  const picture_url = document.getElementById("recipePicture").value;
+  const description = document.getElementById("recipeDescription").value;
+
+  const ingredientElements = document.getElementsByClassName("ingredient");
+  const ingredients = [];
+
+  for (let i = 0; i < ingredientElements.length; i++) {
+      const name = ingredientElements[i].querySelector(".ingredient-name").value;
+      const amount = ingredientElements[i].querySelector(".ingredient-amount").value;
+      ingredients.push({ name: name, amount: amount });
+  }
+
+  const newRecipe = { id: Date.now(), title, picture_url, description, ingredients };
+  recipes.push(newRecipe);
+  displayRecipes(recipes);
+});
+
+// Add Ingredient
+document.getElementById("addIngredientButton").addEventListener("click", function () {
+  const div = document.createElement("div");
+  div.className = "ingredient";
+  div.innerHTML = `
+    <label>Ingredient: <input type="text" class="ingredient-name" required></label>
+    <label>Amount: <input type="text" class="ingredient-amount" required></label>
+  `;
+  document.getElementById("ingredientsList").appendChild(div);
+});
+
+// Search Recipes
+document.getElementById("search1").addEventListener("click", function () {
+  const searchTerm = document.getElementById("searchInput").value.toLowerCase();
+  const filtered = recipes.filter(recipe => recipe.title.toLowerCase().includes(searchTerm));
+  displayRecipes(filtered);
+});
+
+// Sort Recipes
+document.getElementById("sortButton").addEventListener("click", function () {
+  const sorted = [...recipes].sort((a, b) => a.ingredients.length - b.ingredients.length);
+  displayRecipes(sorted);
+});
+// Timer
+function startCookingTimer() {
+const cookingTimeInput = document.getElementById('cookingTime').value;
+const message = document.getElementById('cookingMessage');
+
+if (!cookingTimeInput || cookingTimeInput.length === 0 || Number(cookingTimeInput) <= 0) {
+  message.textContent = "Please enter a valid positive number.";
+  return;
+}
+
+const cookingTime = Number(cookingTimeInput);
+message.textContent = `Cooking timer started for ${cookingTime} seconds.`;
+
+setTimeout(() => {
+  alert("Cooking time is up!");
+  message.textContent = "Time's up! Your dish is ready.";
+}, cookingTime * 1000);
+}
+
+document.getElementById('startTimerButton').addEventListener('click', startCookingTimer);
+
+(function trackPageTime() {
+let secondsSpent = 0;
+
+function updatePageTime() {
+  secondsSpent++;
+  document.getElementById('pageTimer').textContent = `${secondsSpent} seconds`;
+}
+
+setInterval(updatePageTime, 1000);
+})();
+
+
+//fetch data
+async function fetchData() {
```

Comment:
I think this part was changed in one of the previous PRs where we agreed to split fetching data from displaying it